src/test/selectionRange.test.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import 'mocha';
+import * as assert from 'assert';
+import { TextDocument } from 'vscode-languageserver-types';
+import { getApplicableRanges } from '../services/htmlSelectionRange';
+
+function assertRanges(lines: string[] | string, expected: number[][]): void {
+	let content: string = '';
+	if (Array.isArray(lines)) {
+		content = lines.join('\n');
+	} else {
+		content = lines;
+	}
+	let message = `Test ${content}`;
+
+	let offset = content.indexOf('|');
+	content = content.substr(0, offset) + content.substr(offset + 1);
+
+	const document = TextDocument.create('test://foo/bar.html', 'html', 1, content);
+	const actualRanges = getApplicableRanges(document, document.positionAt(offset));
+
+	message += `\n${JSON.stringify(actualRanges)} should equal to ${JSON.stringify(expected)}`;
+	assert.deepEqual(actualRanges, expected, message);
+}
+
+suite('HTML SelectionRange', () => {
+	test('Basic', () => {
+		assertRanges('<div|>foo</div>', [[1, 4], [0, 14]]);
```

Comment:
shouldn't [1,4] be [1,3]?
For readability, maybe format the expected ranges like `[ "1:div", "0:<div>foo</div>"]`